docs/design/persistent-volume-provisioning.md

```diff
@@ -0,0 +1,129 @@
+# Persistent Volume Provisioning
+
+This document proposes a model for dynamically provisioning [Persistent Volumes](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/persistent-storage.md)
+
+### tl;dr
+
+One new API kinds:
+
+A `PersistentVolumeController` (PVC) is a storage resource provisioned by an administrator.  PVCs do not have a namespace.  Like a ReplicationController maintains a minimum number of "replicas" of a pod in the system, so a PersistentVolumeController maintains a minimum number of "replicas" of a certain type/size of PersistentVolume available in the system.
+
+One new system component:
+
+`PersistentVolumeControllerManager` is a singleton control loop running in master that manages all PVControllers in the system.  The PVCM reconciles the current supply of available PersistentVolumes in the system with the desired levels according to the PVControllers.   This process is similar to the ```ReplicationManager``` that manages ReplicationControllers.
+
+Two new persistent volume functions:
+
+The PersistentVolumeSource interface requires ```Create``` and ```Recycle``` functions to support the dynamic creation and reclamation of persistent volumes.  Each storage provider implements its own PersistentVolumeSource interface.
+
+Caveat:  Not all persistent volumes will support dynamic provisioning.
```

Comment:
So long as a plugin can call exec and the admin creating the PVControllers can specify what to call, I suppose the caveat should be "some assembly required" for some types of volumes.